### DISCUSSION TITLE
Booting on Shell Plus 1 fails with invalid header

### DISCUSSION THREAD

**Discussion by l-jonas** [arendst/Tasmota#15507] [Q&A] [answered]
What I did:

```
$ wget https://ota.tasmota.com/tasmota32/release/tasmota32solo1.bin
$ esptool.py --chip esp32 --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dout --flash_size detect 0x0 tasmota32solo1.bin
esptool.py v3.3
Found 1 serial ports
Serial port /dev/ttyUSB0
Connecting....
Chip is ESP32-U4WDH (revision 3)
Features: WiFi, BT, Single Core, Embedded Flash, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: XX:XX:XX:XX:XX:XX
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Auto-detected Flash size: 4MB
Flash will be erased from 0x00000000 to 0x0015afff...
Compressed 1420784 bytes to 995913...
Wrote 1420784 bytes (995913 compressed) at 0x00000000 in 16.1 seconds (effective 707.0 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
```

After this, no tasmota WiFi is created. On the serial port, the following is printed on boot:

```
rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
invalid header: 0x3f400d74
invalid header: 0x3f400d74
invalid header: 0x3f400d74
invalid header: 0x3f400d74
invalid header: 0x3f400d74
invalid header: 0x3f400d74
invalid header: 0x3f400d74
ets Jul 29 2019 12:21:46

rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
invalid header: 0x3f400d74
invalid header: 0x3f400d74
invalid header: 0x3f400d74
invalid header: 0x3f400d74
invalid header: 0x3f400d74
invalid header: 0x3f400d74
invalid header: 0x3f400d74
ets Jul 29 2019 12:21:46
```

This text is printed again and again.

I don't know if that's related to the chip revision or if the current ROM is defect. How can I further troubleshoot this?

**Comment by sfromis** [comment, marked as answer]
You did not do a valid flashing, as of the Tasmota docs about [how to flash ESP32](https://tasmota.github.io/docs/ESP32/#flashing).
For flashing a single file at offset 0, you should use a `.factory.bin`

**Comment by l-jonas** [comment]
That's a good hint.

The irritating point is that <https://templates.blakadder.com/shelly_plus_1.html> says one must flash [tasmota32solo1.bin](http://ota.tasmota.com/tasmota32/release/tasmota32solo1.bin). The ``tasmota32solo1.factory.bin`` seems to work. The factory-Version is not available at <https://ota.tasmota.com/tasmota32/release/> but at <https://github.com/arendst/Tasmota/releases>. This is a bit difficult for beginners.

**Comment by Jason2866** [comment]
Easiest is to use Tasmota WebInstaller as recommended in the Tasmota Documentation.

**Comment by argoncoast** [comment]
Thanks for posting this, I hit exactly the same issue and whilst there's a sample `esptool.py` command on the linked documentation posted by @sfromis it does say on templates page to use `tasmota32solo1.bin`.

The documentation page linked doesn't explain why the factory image must be used that I can see and I hit exactly the same issue today and after a lot of searching found this page.

Whilst the Tasmota Webinstaller may have been a better option (I can't say as I haven't used it) for someone like myself who's running Linux on the desktop, without a chrome based browser, and used to using the terminal I agree it might be worth making the need to use the factory image a bit more obvious.

**Comment by Jason2866** [comment]
The documentation is Open Source too. You are welcome to contribute. Just click the pencil in the upper right corner.

**Reply by argoncoast** [reply]
That's a fair answer and I've got no issues with submitting a PR, but perhaps what I'm trying to say is "What is the factory image and why do we need to use it?" 

From reading on other sites I *think* it has all the bootloader, partition info and files required and the non-factory image doesn't, which kind of explains why it needs a 0 offset.

I there any other relevant information that should be documented or is the above all the necessary documentation?

**Reply by Jason2866** [reply]
Thats all.
